ledger/participant-state/kvutils/tools/src/test/scala/ledger/participant/state/kvutils/tools/integritycheck/StateUpdateExporterSpec.scala

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.tools.integritycheck
+
+import java.io.Writer
+import java.nio.file.Path
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import akka.testkit.TestKit
+import com.daml.ledger.api.health.HealthStatus
+import com.daml.ledger.participant.state.v1.{LedgerInitialConditions, Offset, Update}
+import com.daml.lf.data.Time
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpecLike
+
+import scala.concurrent.ExecutionContext
+
+class StateUpdateExporterSpec
+    extends TestKit(ActorSystem(classOf[StateUpdateExporterSpec].getSimpleName))
+    with AsyncWordSpecLike
+    with Matchers
+    with MockitoSugar
+    with ArgumentMatchersSugar
+    with BeforeAndAfterAll {
+  import StateUpdateExporterSpec._
+
+  private[this] implicit val materializer: Materializer = Materializer(system)
+  private[this] implicit val executionContext: ExecutionContext = materializer.executionContext
+
+  override def afterAll(): Unit = {
+    TestKit.shutdownActorSystem(system)
+  }
+
+  "write" should {
+    "write updates" in {
+      val mockWriter = mock[Writer]
+      StateUpdateExporter
+        .write(
+          aReadService,
+          aReadService,
+          _ => mockWriter,
```

Comment:
Can you enhance this test to use two `Writer` objects differentiated by path, to ensure that the path is used correctly?

Reply:
done